definitions/README.md

```diff
@@ -2,3 +2,84 @@
 
 This directory contains definitions for specific metrics, data sources, dimensions or segments for the platform they target.
 These definitions can be referenced in project specific configuration files without having to be redefined.
+
+Metric definitions can be validated using the following Python script in Google Colab:
+
+First, ensure that `mozanalysis` is available:
+
+```shell
+!pip install mozanalysis -qq
+```
+
+Second, setup the environment:
+
+```python3
+import pandas as pd
+
+from google.cloud import bigquery
+from google.colab import auth
+from mozanalysis.config import ConfigLoader
+from textwrap import dedent
+
+auth.authenticate_user()
+
+def fetch_metric_hub_definition(start_date, end_date, metric_slug, app_name, bq_project):
+    start_date = pd.to_datetime(start_date).date()
+    end_date = pd.to_datetime(end_date).date()
+
+    # Set useful attributes based on the Metric Hub definition
+    metric = ConfigLoader.get_metric(metric_slug=metric_slug, app_name=app_name)
+    submission_date_column = metric.data_source.submission_date_column
+
+    # Modify the metric source table string so that it formats nicely in the query.
+    from_expression = metric.data_source._from_expr.replace("\n", "\n" + " " * 15)
+
+    query = dedent(
```

Comment:
What about adding the query builder as a method inside of `DataSource`, something like `DataSource.build_validation_query(start_date: str, end_date: str) -> str`. Then we can expose that through the `Metric` through something like: 

```python3
# in Metric class
def build_validation_query(self, start_date: str, end_date: str) -> str:
    return self.data_source.build_validation_query(start_date, end_date)
```

In the `DataSource` method, we can handle modifying the from expression like how `build_query` [does it](https://github.com/mozilla/mozanalysis/blob/main/src/mozanalysis/metrics.py#L177). Additionally, that gives us a separate method to unit test. Right now, the logic is hidden within this larger function.

Reply:
If we do that, this function can be simplified to: 

```python3
metric = ConfigLoader.get_metric(metric_slug=metric_slug, app_name=app_name)
query = metric.build_validation_query(start_date, end_date)
df = bigquery.Client(project=bq_project).query(query).to_dataframe()
df[submission_date_column] = pd.to_datetime(df[submission_date_column]).dt.date
return df
```

Reply:
I really, really like the idea of having an attribute on Metric that provides a query. I don't know if it's difficult to do in the general case, but it's something I think could be useful outside of this case. For example, there's a [class in the forecasting code](https://github.com/mozilla/docker-etl/blob/main/jobs/kpi-forecasting/kpi_forecasting/metric_hub.py) whose main goal is to build a query from Metric Hub components.